share/completions/make.fish

```diff
@@ -12,9 +12,13 @@ function __fish_print_make_targets --argument-names directory file
     if make --version 2>/dev/null | string match -q 'GNU*'
         # https://stackoverflow.com/a/26339924
         make $makeflags -pRrq : 2>/dev/null |
-            awk -F: '/^# Files/,/^# Finished Make data base/ {
+            awk -F: -v 'backslash=\\\\' '/^# Files/,/^# Finished Make data base/ {
                 if ($1 == "# Not a target") skip = 1;
-                if ($1 !~ "^[#.\t]") { if (!skip) print $1; skip=0 }
+                if ($1 !~ "^[#.\t]" && !is_continuation ) {
+                    if (!skip) print $1;
+                    skip = 0
+                }
+                is_continuation = $0 ~ backslash backslash "$";
```

Comment:
`is_continuation` has two false positives:

```Makefile
t1:
	: \\
target1:
	:

t2:
	:
	# asdf \
target2:
	:
```

one of them we can fix with

```diff
diff --git a/share/completions/make.fish b/share/completions/make.fish
index 035318ccc..ce5abb9af 100644
--- a/share/completions/make.fish
+++ b/share/completions/make.fish
@@ -20,3 +20,3 @@ function __fish_print_make_targets --argument-names directory file
                 }
-                is_continuation = $0 ~ backslash backslash "$";
+                is_continuation = $0 ~ "(" backslash backslash backslash backslash ")*" backslash backslash "$";
             }' 2>/dev/null
```

the other by detecting comments

Reply:
Thanks, should be good now.